### ISSUE TITLE
PHP 8.0 Deprecation with parameter order causing error: Required parameter $array follows optional parameter $glue in ArrayHelper.php:23

### ISSUE THREAD

**Issue by chadwells** [verbb/formie#1608]
### Describe the bug

We are running into some logged errors that end up breaking parts of our staging site. The error is apparently related to [this PHP deprecation (the first one listed)](https://www.php.net/manual/en/migration80.deprecated.php). The logged error is:

```
2023-10-24 10:47:22 [-][1981021][501175cc9c598ab3a828c3d2b41a5d15][error][yii\base\ErrorException:8192] yii\base\ErrorException: Required parameter $array follows optional parameter $glue in /home/fct/staging/vendor/verbb/formie/src/helpers/ArrayHelper.php:23
Stack trace:
#0 /home/fct/staging/vendor/craftcms/cms/src/web/ErrorHandler.php(87): yii\base\ErrorHandler->handleError(8192, 'Required parame...', '/home/fct/stagi...', 23)
#1 /home/fct/staging/vendor/composer/ClassLoader.php(571): craft\web\ErrorHandler->handleError(8192, 'Required parame...', '/home/fct/stagi...', 23)
#2 /home/fct/staging/vendor/composer/ClassLoader.php(571): include()
#3 /home/fct/staging/vendor/composer/ClassLoader.php(428): Composer\Autoload\includeFile('/home/fct/stagi...')
#4 /home/fct/staging/vendor/verbb/formie/src/elements/Form.php(1127): Composer\Autoload\ClassLoader->loadClass('verbb\\formie\\he...')
#5 /home/fct/staging/vendor/twig/twig/src/Extension/CoreExtension.php(1570): verbb\formie\elements\Form->getFieldByHandle('email')
#6 /home/fct/staging/vendor/craftcms/cms/src/helpers/Template.php(106): twig_get_attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(verbb\formie\elements\Form), 'getFieldByHandl...', Array, 'method', false, false)
#7 /home/fct/staging/storage/runtime/compiled_templates/47/47f71e48903a6fb40da87df27adce3fbd541e827f3b8d974c651670d30e27ada.php(206): craft\helpers\Template::attribute(Object(craft\web\twig\Environment), Object(Twig\Source), Object(verbb\formie\elements\Form), 'getFieldByHandl...', Array, 'method')
#8 /home/fct/staging/vendor/twig/twig/src/Template.php(405): __TwigTemplate_cfd1e91e99c8fdeb695d1d776ce8e81d6295ad726bfac642780272fed0920f05->doDisplay(Array, Array)
#9 /home/fct/staging/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#10 /home/fct/staging/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#11 /home/fct/staging/vendor/twig/twig/src/Extension/CoreExtension.php(1290): Twig\Template->render(Array)
#12 /home/fct/staging/storage/runtime/compiled_templates/16/1631add7d6c55ee93a3b76d42dcc61d8eb6805d52846189e9d86a1ad37feb076.php(342): twig_include(Object(craft\web\twig\Environment), Array, '_partials/foote...', Array, false)
#13 /home/fct/staging/vendor/twig/twig/src/Template.php(405): __TwigTemplate_e637dbd3b6f51a81069a9010527cc1c8fdcbddf9313b014114125c1f964eef97->doDisplay(Array, Array)
#14 /home/fct/staging/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#15 /home/fct/staging/storage/runtime/compiled_templates/58/588df789a2978f3029800b7c1f203e5729ae54a9f0628fb2b6a8f1211a343e0e.php(69): Twig\Template->display(Array, Array)
#16 /home/fct/staging/vendor/twig/twig/src/Template.php(405): __TwigTemplate_502c1129e71bdbc16df32c9ae436dd08d202a690839a8cd9a3efb9893d5dcf3c->doDisplay(Array, Array)
#17 /home/fct/staging/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#18 /home/fct/staging/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#19 /home/fct/staging/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#20 /home/fct/staging/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#21 /home/fct/staging/vendor/craftcms/cms/src/web/View.php(408): Twig\Environment->render('_pages/_types/a...', Array)
#22 /home/fct/staging/vendor/craftcms/cms/src/web/View.php(461): craft\web\View->renderTemplate('_pages/_types/a...', Array)
#23 /home/fct/staging/vendor/craftcms/cms/src/web/Controller.php(216): craft\web\View->renderPageTemplate('_pages/_types/a...', Array, 'site')
#24 /home/fct/staging/vendor/craftcms/cms/src/controllers/TemplatesController.php(102): craft\web\Controller->renderTemplate('_pages/_types/a...', Array)
#25 [internal function]: craft\controllers\TemplatesController->actionRender('_pages/_types/a...', Array)
#26 /home/fct/staging/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#27 /home/fct/staging/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#28 /home/fct/staging/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('render', Array)
#29 /home/fct/staging/vendor/craftcms/cms/src/web/Application.php(295): yii\base\Module->runAction('templates/rende...', Array)
#30 /home/fct/staging/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('templates/rende...', Array)
#31 /home/fct/staging/vendor/craftcms/cms/src/web/Application.php(280): yii\web\Application->handleRequest(Object(craft\web\Request))
#32 /home/fct/staging/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#33 /home/fct/staging/web/index.php(12): yii\base\Application->run()
#34 {main}

```

From what I've read, it can be relatively easily resolved by removing the default value on the first parameter of `src/helpers/ArrayHelper.php:recursiveImplode` 

### Steps to reproduce

What's a bit strange is we've only really been seeing this on our Staging site, but it is causing enough of an issue that we cannot view some newly developed features of our site because of this issue. It's hard for me to give a specific set of steps to reproduce in this case, other than going to this page: https://staging.fireclaytile.com/account/dashboard The only use of formie there is in the page footer.  This is on Craft 3.9.3, Formie 1.6.33, and PHP 8.0.30.

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

### Craft CMS version

3.9.3

### Plugin version

1.6.33

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-3 as 1.6.34"`.

**Comment by engram-design** [comment]
Fixed in [1.6.35](https://github.com/verbb/formie/releases/tag/1.6.35)
